File: process_logo.py
import sys
import os

try:
    from PIL import Image
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "Pillow"])
    from PIL import Image

def remove_background(input_path, output_path):
    img = Image.open(input_path).convert("RGBA")
    datas = img.getdata()
    
    newData = []
    # Make black background transparent
    for item in datas:
        # Dark pixels with some tolerance
        if item[0] < 25 and item[1] < 25 and item[2] < 25:
            # Change alpha to 0
            newData.append((0, 0, 0, 0))
        else:
            newData.append(item)
            
    img.putdata(newData)
    img.save(output_path, "PNG")
    print(f"Saved transparent logo to {output_path}")

input_img = r"c:\Users\Juan\Documents\Projetos Starken\alces-barbearia\mobile\logo\logo.png"
output_img = r"c:\Users\Juan\Documents\Projetos Starken\alces-barbearia\mobile\assets\images\logo_transparent.png"

remove_background(input_img, output_img)

File: process_logo.js
const { Jimp } = require('jimp');

async function removeBackground() {
  try {
    const inputPath = 'mobile/logo/logo.png';
    const outputPath = 'mobile/assets/images/logo.png';
    
    // Ler a imagem
    const image = await Jimp.read(inputPath);
    
    // Loop pelos pixels
    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
      const r = this.bitmap.data[idx];
      const g = this.bitmap.data[idx + 1];
      const b = this.bitmap.data[idx + 2];
      
      if (r < 25 && g < 25 && b < 25) {
        this.bitmap.data[idx + 3] = 0;
      }
    });

    image.write(outputPath);
    console.log('Fundo preto removido e imagem transparente gerada com sucesso em: ' + outputPath);
  } catch (error) {
    console.error('Erro ao processar imagem:', error);
  }
}

removeBackground();
